assets/js/data/cart/push-changes.ts

```diff
@@ -65,39 +63,91 @@ const isAddressDirty = <
 	);
 };
 
+type BaseAddressKey = keyof CartBillingAddress | keyof CartShippingAddress;
+
+const getDirtyKeys = < T extends CartBillingAddress | CartShippingAddress >(
+	// An object containing all previous address information
+	previousAddress: T,
+	// An object containing all address information.
+	address: T
+): BaseAddressKey[] => {
+	const previousAddressKeys = Object.keys(
+		previousAddress
+	) as BaseAddressKey[];
+
+	return previousAddressKeys.filter( ( key ) => {
+		return previousAddress[ key ] !== address[ key ];
+	} );
+};
+
 /**
  * Local cache of customerData used for comparisons.
  */
 let customerData = <CustomerData>{
 	billingAddress: {},
 	shippingAddress: {},
 };
+
 // Tracks if customerData has been populated.
 let customerDataIsInitialized = false;
 
 /**
  * Tracks which props have changed so the correct data gets pushed to the server.
  */
-const dirtyProps = {
-	billingAddress: false,
-	shippingAddress: false,
+const dirtyProps = <
+	{
+		billingAddress: BaseAddressKey[];
+		shippingAddress: BaseAddressKey[];
+	}
+>{
+	billingAddress: [],
+	shippingAddress: [],
 };
 
 /**
  * Function to dispatch an update to the server. This is debounced.
  */
 const updateCustomerData = debounce( (): void => {
 	const { billingAddress, shippingAddress } = customerData;
+	const validationStore = select( VALIDATION_STORE_KEY );
 	const customerDataToUpdate = {} as Partial< BillingAddressShippingAddress >;
 
-	if ( dirtyProps.billingAddress ) {
-		customerDataToUpdate.billing_address = billingAddress;
-		dirtyProps.billingAddress = false;
+	// Before we push anything, we need to ensure that the data we're pushing (dirty fields) are valid, otherwise we will
+	// abort and wait for the validation issues to be resolved.
+	const invalidProps = [
```

Comment:
The logic here validates the **changed** props. If a **changed** prop has a validation error, pushing is aborted until the customer fixes the changed prop. This avoids invalid pushes of data.